### PR TITLE
refactor: use zx clone (dax) for executing commands and making requests

### PR DESCRIPTION
**Pull request by dsherret**
This simplifies a few things.

### CODE REVIEW

File: crate.ts [hunk 1]
```diff
@@ -79,7 +73,7 @@ export class Crate {
   }
 
   async setVersion(version: string) {
-    console.log(`Setting ${this.name} to ${version}...`);
+    $.logTitle(`Setting ${this.name} to ${version}...`);
```
Comment (dsherret) [inline crate.ts:76]: This will output with the first word as bold green.

Comment (dsherret) [inline crate.ts:76]: This will output with the first word as bold green.

File: crate.ts [hunk 2]
```diff
@@ -52,11 +46,11 @@ export class Crate {
 
   /** Prompts the user how they would like to patch and increments the version accordingly. */
   async promptAndTryIncrement() {
-    console.log(`${this.name} is on ${this.version}`);
+    $.log(`${this.name} is on ${this.version}`);
     const versionIncrement = getVersionIncrement();
     if (versionIncrement != null) {
       await this.increment(versionIncrement);
-      console.log(`Set version to ${this.version}`);
+      $.log(`  Set version to ${this.version}`);
```
Comment (dsherret) [inline crate.ts:53]: `$.log` is preferrable because it allows being indented if necessary by something upstream.

**Review by bartlomieju — APPROVED** [review]
LGTM!

**Pull request merged**